client.go

```diff
@@ -148,12 +148,20 @@ func (c *Client) defaultTimeout(ctx context.Context) time.Duration {
 	return time.Until(deadline) / time.Duration(l)
 }
 
+func cloneHTTPClient(old *http.Client) *http.Client {
+	client := *old
+	transport, ok := client.Transport.(*http.Transport)
+	if !ok {
+		panic("old client's transport is not expected type")
```

Comment:
hmm...I'm concerned about this panic. I know we weren't even checking the type assert before, and it could have implicitly caused a panic already, but could we introduce a more graceful way to handle the bad type assert? Maybe a return error or something similar?

Reply:
I don't think its worth it, changing to an error return is going to have wide effect when done for `registerASRRProvider` and probably others (bubbling the error up). Its never really going to be a problem either since we aren't really using different implementations. If it ever changes then it'll be caught early at startup instead of later at runtime.

Not a whole lot of benefit imo.

Reply:
This does seem like an exceptional case, can the panic message include a clearer message, along the lines of "expected a http client of the http.Transport type, got $blah instead"?

Reply:
I've updated the panic message